pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -519,6 +524,20 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	return 0
 }
 
+func waitUntilNodeAgentReady(ctx *zedagentContext, running *time.Ticker) {
+	log.Functionf("waitUntilNodeAgentReady()")
+	for !ctx.GetNodeAgentStatus {
+		select {
+		case change := <-ctx.getconfigCtx.subNodeAgentStatus.MsgChan():
+			ctx.getconfigCtx.subNodeAgentStatus.ProcessChange(change)
+			ctx.GetNodeAgentStatus = true
```

Comment:
I think it is safer to set `ctx.GetNodeAgentStatus = true` from inside of `handleNodeAgentStatusImpl`. There are various pubsub control messages. The first one received may not necessary mean that something was published and that it was received by the subscriber.

Reply:
You can maybe also check if RebootReason is available (`zedagentContext.rebootReason != ""`), and only then break from the for loop. So perhaps no additional bool flag needed.

Reply:
(but then I would rename the function to something like `waitForRebootReason`)

Reply:
Make sense! I will rework it. Thanks for the recommendation.

Reply:
Actually context take pretty a lot info from NodeAgentStatus:
https://github.com/lf-edge/eve/blob/5849cb0bc5f44d3c95985656a0f4f8df441490c1/pkg/pillar/cmd/zedagent/zedagent.go#L2340-L2347
And I think it makes sense to make the function more generic than just "waitForBootReason". Now I see the problem with the bootReason field, but potentially there are others.

Reply:
LGTM